integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/utils.py

```diff
@@ -143,78 +403,18 @@ def _parse_streaming_response(
-
+    chunks: List[StreamingChunk] = []
     for event in response_stream:
-        if "contentBlockStart" in event:
-            # Reset accumulators for new message
-            current_content = ""
-            current_tool_call = None
-            block_start = event["contentBlockStart"]
-            if "start" in block_start and "toolUse" in block_start["start"]:
-                tool_start = block_start["start"]["toolUse"]
-                current_tool_call = {
-                    "id": tool_start["toolUseId"],
-                    "name": tool_start["name"],
-                    "arguments": "",  # Will accumulate deltas as string
-                }
-
-        elif "contentBlockDelta" in event:
-            delta = event["contentBlockDelta"]["delta"]
-            if "text" in delta:
-                delta_text = delta["text"]
-                current_content += delta_text
-                streaming_chunk = StreamingChunk(content=delta_text, meta={})
-                streaming_callback(streaming_chunk)
-            elif "toolUse" in delta and current_tool_call:
-                # Accumulate tool use input deltas
-                current_tool_call["arguments"] += delta["toolUse"].get("input", "")
-
-        elif "contentBlockStop" in event:
-            if current_tool_call:
-                # Parse accumulated input if it's a JSON string
-                try:
-                    input_json = json.loads(current_tool_call["arguments"])
-                    current_tool_call["arguments"] = input_json
-                except json.JSONDecodeError:
-                    # Keep as string if not valid JSON
-                    pass
-
-                tool_call = ToolCall(
-                    id=current_tool_call["id"],
-                    tool_name=current_tool_call["name"],
-                    arguments=current_tool_call["arguments"],
-                )
-                replies.append(ChatMessage.from_assistant("", tool_calls=[tool_call], meta=base_meta.copy()))
-            elif current_content:
-                replies.append(ChatMessage.from_assistant(current_content, meta=base_meta.copy()))
-
-        elif "messageStop" in event:
-            # Update finish reason for all replies
-            for reply in replies:
-                reply.meta["finish_reason"] = event["messageStop"].get("stopReason")
-
-        elif "metadata" in event:
-            metadata = event["metadata"]
-            # Update usage stats for all replies
-            for reply in replies:
-                if "usage" in metadata:
-                    usage = metadata["usage"]
-                    reply.meta["usage"] = {
-                        "prompt_tokens": usage.get("inputTokens", 0),
-                        "completion_tokens": usage.get("outputTokens", 0),
-                        "total_tokens": usage.get("totalTokens", 0),
-                    }
-
+        streaming_chunk = _convert_event_to_streaming_chunk(event=event, model=model)
```

Comment:
I'd put somewhere here a short pydoc comment or a note to ourselves here I'd put that we receive these chunks from Bedrock and sometimes they are guaranteed to have atomic units with specific fields toolUseId, name etc (that's why we are guaranteed to be able to access them from event dict)  yet some blocks will contain only arguments for a function that we need to accumulate.